src/typecheck.cpp

```diff
@@ -240,10 +240,12 @@ void typecheck(TypedAST::RecordAccessExpression* ast, TypeChecker& tc) {
 
 void typecheck(TypedAST::ConstructorExpression* ast, TypeChecker& tc) {
 	typecheck(ast->m_constructor, tc);
-	TypeFunctionId tf = tc.m_core.m_tf_core.find_term(ast->m_constructor->m_value_type);
-	TypeFunctionData& tf_data = tc.m_core.m_type_functions[tf];
 
-	// TODO: match type arguments as well, we only support 0 now
+	auto handle = static_cast<TypedAST::MonoTypeHandle*>(ast->m_constructor);
+	assert(handle && "Error: Value constructor was not handled");
+
+	TypeFunctionId tf = tc.m_core.m_mono_core.find_function(handle->m_value);
+	TypeFunctionData& tf_data = tc.m_core.m_type_functions[tf];
```

Comment:
Is this right? I can't seem to keep track of how typefuncs are stored after the unification refactor

Reply:
A monotype handle stores a MonoId as its value. At the `m_mono_core`, the terms are applications of type functions with some mono arguments, so I use the very handy `find_function` to get the function directly (and not the term). That is a TypeFunctionId, so I fetch its data (that has been separated somewhere else for not being term-like) in the special `m_type_functions` vector.

Reply:
And TypeFunctionId is an index into that array? And not into `m_core.m_tf_core.terms`?